holmes/plugins/toolsets/prometheus/prometheus.py

```diff
@@ -816,3 +825,16 @@ def get_example_config(self):
             prometheus_url="http://robusta-kube-prometheus-st-prometheus:9090"
         )
         return example_config.model_dump()
+
+    def init_config(self, config: dict[str, Any]):
+        if config:
+            self.typed_config = PrometheusConfig(**config)
+        else:
+            prometheus_url = os.environ.get("PROMETHEUS_URL")
+            if not prometheus_url:
+                prometheus_url = self.auto_detect_prometheus_url()
+            self.typed_config = PrometheusConfig(
+                prometheus_url=prometheus_url,
+                headers=add_prometheus_auth(os.environ.get("PROMETHEUS_AUTH_HEADER")),
+            )
+            logging.info(f"Prometheus auto discovered at url {prometheus_url}")
```

Comment:
There is a regression here about error handling when prometheus auto discovery fails.
I understand that `self._is_healthy()` will report a config error but it's a generic message. The failure message that was there before this change `"Unable to auto-detect prometheus. Define prometheus_url in the configuration for tool prometheus/metrics"` is very helpful to users that rely on auto discovery and we should keep it.

Reply:
added auto-detect back.